### PR TITLE
Add channel export API

### PR DESCRIPTION
**Pull request by tbarbugli**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -2150,4 +2153,31 @@ export class StreamChat<
   deleteBlockList(name: string) {
     return this.delete<APIResponse>(`${this.baseURL}/blocklists/${name}`);
   }
+
+  exportChannels(request: Array<ExportChannelRequest>) {
```
Comment (mahboubii) [inline src/client.ts:2157]: this is a server only feature? should we throw error when it's called using client tokens?

File: src/client.ts [hunk 2]
```diff
@@ -2150,4 +2153,31 @@ export class StreamChat<
...
+  }
+
+  getExportChannelStatus(id: string) {
+    return this.get<APIResponse & ExportChannelStatusResponse>(
+      `${this.baseURL}/export_channels/${id}`,
+    );
+  }
+}
+
+function makeExportPayload(v: ExportChannelRequest) {
```
Comment (mahboubii) [inline src/client.ts:2176]: i'd merge this into `client.exportChannels` since it's simple and not used anywhere else

File: src/types.ts [hunk 3]
```diff
@@ -1203,6 +1214,13 @@ export type BlockList = {
   words: string[];
 };
 
+export type ExportChannelRequest = {
+  id: string;
+  type: string;
+  messages_since?: Date;
+  messages_until?: Date;
```
Comment (mahboubii) [inline src/types.ts:1221]: we might want to support string too for ease of use

File: src/types.ts [hunk 4]
```diff
@@ -103,6 +103,17 @@ export type BlockListResponse = BlockList & {
   updated_at?: string;
 };
 
+export type ExportChannelResponse = {
+  task_id: string;
+};
+
+export type ExportChannelStatusResponse = {
```
Comment (mahboubii) [inline src/types.ts:110]: missing ` status` field?

File: src/client.ts (hunk 1 above)
Comment (tbarbugli) [reply src/client.ts:2157]: yes

**Review by mahboubii — APPROVED** [review]

**Pull request merged**